Skin updates

poelzi opened the pull request:
I was putting some love into the deer skin and was able to skin the menubar after some small changes. Since we are skinning, having a default menubar feels like a foreign object, so I consider this 2.3 worthy.

* Deere: option for a larger UV master.
* Indicate green/yellow area in the UV meters
* ~~Allow skinning of menubar~~ #3372 
* ~~Skin Deer Menubar~~ future PR by @ronso0 
* Small adjustments of the alpha value for track colors for better visibility (all skins)


<img width="944" alt="Screenshot_20201111_002045" src="https://user-images.githubusercontent.com/66107/98745396-c966ae80-23b3-11eb-94c5-9bd7f0861bbb.png">

poelzi committed:
[Deer] Add skin option: large uv meter

If the option is enabled, the master UV meter will takeup
as much space as possible. Especially on the 4 deck settings
with all controls enabled, the UV meter is very small.

poelzi committed:
[Deer] Add colored UV meter background

Show in the unlight UV meter which parts are green and yellow
through a darker version of the normal meter.

poelzi committed:
[Skins] Increase the alpha value on track table backgrounds

Slightly increase the alpha value so the track color is better visible.

daschuer commented:
I cannot see a difference in case of Gnome or Unity. 
Can you add also a "Before" Screenshot?

daschuer commented:
I am in doubt that the large VU Meter should be a preference option. The designer should decide.

daschuer commented:
OK there is a difference with Deere:
![grafik](https://user-images.githubusercontent.com/1777442/98809893-9b6a8400-241e-11eb-9dba-112244df2190.png)

poelzi commented:
> I am in doubt that the large VU Meter should be a preference option. The designer should decide.

I differ. Some people might like to have same sized UV meters and this is a simple option that does not much harm.

poelzi commented:
> I cannot see a difference in case of Gnome or Unity.
> Can you add also a "Before" Screenshot?

It depends on your system colors. If you run a dark skin it might look ok, on a light skin, at least on KDE, it looks ugly.

Holzhaus commented:
> > I cannot see a difference in case of Gnome or Unity.
> > Can you add also a "Before" Screenshot?
> 
> It depends on your system colors. If you run a dark skin it might look ok, on a light skin, at least on KDE, it looks ugly.

https://mixxx.org/screenshots/

ronso0 commented on res/skins/Deere/style.qss:
```diff
@@ -2190,3 +2201,28 @@ WEffectSelector {
+  image: url(skin:/icon/ic_library_checkmark.svg);
+}
+
+#MainMenu QMenu::indicator:unchecked {
+  background: none;
+  border: 1px solid rgba(151,151,151,128);
+}
```
Can be set together with `#LibraryContainer QTableView::indicator:unchecked` in line 276

ronso0 commented on res/skins/Deere/style.qss:
```diff
@@ -2190,3 +2201,28 @@ WEffectSelector {
+  padding: 5px 13px 5px 10px;
+}
+
+#MainMenu QMenu::indicator:checked,
+#MainMenu QMenu::indicator:checked:selected {
+  image: url(skin:/icon/ic_library_checkmark.svg);
+}
```
Can be moved up to ```
#LibraryContainer QTableView::indicator:checked,
#LibraryContainer QTableView::indicator:checked:selected
```, only the border colors differ.

ronso0 commented:
> OK there is a difference with Deere:
> ![grafik](https://user-images.githubusercontent.com/1777442/98809893-9b6a8400-241e-11eb-9dba-112244df2190.png)

One reason I didn't start working on styles for DlgtrackInfo is that once you start overwriting native styles you need to take care of  _all_ other styles, too, and make sure they look good on all platforms. Thus my library menu style PR has grown enormously and has taken ages to finish.

ronso0 commented:
Please rebase on 2.3 so crash fixes from #3144 are included and this PR is testable.

ronso0 commented:
> > I am in doubt that the large VU Meter should be a preference option. The designer should decide.
> 
> I differ. Some people might like to have same sized UV meters and this is a simple option that does not much harm.

@Be-ing ??

poelzi commented:
@ronso0 I felt your pain yesterday. So, maybe as a quick workaround we could just set the colors of the menus and a border.
With changing check symbols the pain started which might have to do with using SVG as icons, or at least, does not make it easier.

poelzi commented:
This is how 2.3 Deer looks with KDE on a light skin.
<img width="1920" alt="Screenshot_20201111_143147" src="https://user-images.githubusercontent.com/66107/98817628-aaefca00-242a-11eb-9481-453bea4c58f0.png">

ronso0 commented:
> @ronso0 I felt your pain yesterday. So, maybe as a quick workaround we could just set the colors of the menus and a border.
> With changing check symbols the pain started which might have to do with using SVG as icons, or at least, does not make it easier.

maybe take a look at a system theme and check how menus and checkboxes are styled there.
I doubt it fails because of SVGs, those work in the track menu. You may need to set the indicator size explicitely in css.

poelzi commented:
@ronso0 do you have some time helping on this one ? I have so many open points that this styling thing drives me nuts. You wrote that you did some styling of the menus already ?

ronso0 commented:
I think it'll take an entire evening to look into this and supply comprehensive styles, let alone applying this to all skins.
Unfortunately I don't have time for this until sunday.

poelzi commented:
@ronso0 even if it takes you 2 weeks, I would be super happy getting your skilled styling on board here. I tried different things this evening but could not get something look nice everywhere :(

ronso0 commented:
Are you asking just about the styling or about purging quirks like the one @daschuer experienced?

ronso0 commented:
@poelzi 
I suggest you split this up into smaller chunks to get the finished parts merged
* Deere design
* track color opacity
* menubar styling, c++ requirements (no skin changes): we need to check what happens on macOS, Windows and various linux DEs, and since we're short on active Win/macOS testers it would be better to have style-capable builds available and users can test stylesheets without having to build Mixxx on their own
* WIP menubar stylesheets

Be-ing commented:
I agree with @daschuer and @Holzhaus. I am not a fan of the big meter option in Deere. It looks out of place to me and I don't think the option is really helpful. Having more options is not inherently good.

ronso0 commented:
> I agree with @daschuer and @Holzhaus. I am not a fan of the big meter option in Deere. It looks out of place to me and I don't think the option is really helpful.

I think it could be a useful addition and we should make it the default.
But it's to big IMO compared to the channel meters. If we limit it to half the height of the mixer it looks good.
Also let's include the colored meter backpaths.

ronso0 commented:
> * menubar styling, c++ requirements (no skin changes): we need to check what happens on macOS, Windows and various linux DEs, and since we're short on active Win/macOS testers it would be better to have style-capable builds available and users can test stylesheets without having to build Mixxx on their own

IMO we should definitively pick that for 2.3 and subsequently style the menubars in point releases.
This feature has been requested for ages.

Be-ing commented:
> IMO we should definitively pick that for 2.3 and subsequently style the menubars in point releases.

I think that effort would be better spent getting rid of the menu bar.

ronso0 commented:
> > IMO we should definitively pick that for 2.3 and subsequently style the menubars in point releases.
> 
> I think that effort would be better spent getting rid of the menu bar.

There's already the next step in progress #3189 getting rid of the menu bar. After that we'll have to rework The Menu (however that will look like) and make it accessible for screen readers.
That'll take ages, and 2.3 will be the stable version for ~2 years :: looking at our release tempo o_o`
That said, it doesn't make much sense to me accepting the current situation while the styling groundwork is already done here.

@poelzi I you don't have time to split off the menubar theming prerequisities I'll open a separate PR with that commit and finish it in the next weeks.

ronso0 commented:
I picked up 851adbe94ba09be622346058d071d54fee102861 for #3372

ronso0 commented:
Alright, I managed to adjust sizes and margins of the menu item, checkboxes & icons by using `em` 
![image](https://user-images.githubusercontent.com/5934199/100610457-0ddcce80-3310-11eb-881a-671096d055d4.png)
PR coming soon...

At first I was puzzled why the font-sizes wouldn't react to OS font size changes, then I realized it would only obey font settings made in `qtconfig` / `Qt5 Settings` (thus the italic font above).
I think we should not mess with the menu font settings since those are important accessibility parameters users need to be able to adjust by themselves (and we don't have to worry about the layout as we have to in the Mixxx skins), so either omitting menu font sizes in the stylesheets or setting them to `1em` ensures readable menus and should not create accessibility regressions.

By setting some aburd font I also noticed that Mixxx font settings are not complete: Qt config could screw up GUI buttons, labels, table headers etc. I think there are only a few adjustments missing, so I guess I can fix all of them for 2.3, too.

poelzi commented:
@ronso0 I removed the skin styling commits here and only left the large VU meter option in deer, the dimmed background image in VU and the increase in background color in the libraries. I adjusted them individually depending on how clear the background is visible.

ronso0 commented:
re large VU meter in Deere:
some tweaking is necessary to align it with some channel controls. https://github.com/ronso0/mixxx/commit/c9dc5002f2cbb1cfd5455b0721b047dd48a4f951 shoud do the trick.

re track colors:
LGTM, could you also increase it in Shade?

ronso0 commented:
btw I made some good progress with the menubar styling (Deere only as of now).
I hope to finish that and apply the styles to the other skins this week.

ronso0 commented:
> re large VU meter in Deere:
> some tweaking is necessary to align it with some channel controls. [ronso0@c9dc500](https://github.com/ronso0/mixxx/commit/c9dc5002f2cbb1cfd5455b0721b047dd48a4f951) shoud do the trick.
> 
> re track colors:
> LGTM, could you also increase it in Shade?

btw to keep the commit history clean please make use of `git --fixup ...` and `git rebase -i --autosquash` for this PR because a) it is rather small and b) more about the effect than about reviewable code.
thx!

poelzi commented:
@ronso0 I tried your patch but the spacing on top just did not make sense for me. The purpose of the large VU option is, to use as much space as possible for the main VU meter. Only the spacer on the bottom for crossfader assignment was required otherwise it looked strange.

ronso0 commented:
@poelzi I'll take another look at this during the weekend.

ronso0 commented:
@poelzi I know the Deere mixer can be a pita to work with but in this branch the alignment of the large Main VU is horrible in some configurations. We should not sacrifice asthetics just for having a large VU meter.
Please try to fix that somehow with conditional spacers.
![image](https://user-images.githubusercontent.com/5934199/103588165-e36eca00-4ee8-11eb-98e1-64d8f4197074.png)
![image](https://user-images.githubusercontent.com/5934199/103588195-f2557c80-4ee8-11eb-851c-2c413f966457.png)

poelzi commented:
I was not able to reproduce this on my branch, it always looks ok

ronso0 commented:
this is the config I refer to: 4 decks, separate waveforms
![image](https://user-images.githubusercontent.com/5934199/107887679-0f2d9880-6f08-11eb-84ba-6bc445e26a1a.png)
how does that look like for you?

please just add some spacers for this config and we can merge.

ronso0 committed:
Deere: align large main VU in 4-deck mode

ronso0 commented:
with https://github.com/ronso0/mixxx/commit/95db5d72178c3d2b02a739eb8b6e71db04ad6de4 it looks like this, there's an offset only when there's no quick effect configured
![image](https://user-images.githubusercontent.com/5934199/107889250-93384e00-6f11-11eb-84dd-d99afd6fc337.png)

pick this, double-check that it looks okay to you and then we can merge.

ronso0 commented on res/skins/Deere/mixer_column_master_vu_4decks.xml:
```diff
@@ -8,86 +8,60 @@
+              <SingletonContainer>
+                <ObjectName>StereoVUMeterMaster</ObjectName>
+              </SingletonContainer>
+            </Children>
```
https://github.com/ronso0/mixxx/commit/95db5d72178c3d2b02a739eb8b6e71db04ad6de4

poelzi committed:
Merge branch '2.3' of https://github.com/mixxxdj/mixxx into skin-updates-menustyling

poelzi commented:
@ronso0 done, looks good here

ronso0 commented:
Okido, the rest LGTM
Thanks!
